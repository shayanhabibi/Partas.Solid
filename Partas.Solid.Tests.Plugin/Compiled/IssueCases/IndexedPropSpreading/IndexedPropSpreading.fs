module Partas.Solid.Tests.IssueCases.IndexedPropSpreading

open Partas.Solid
open Fable.Core

[<Import("splitProps", "solid-js")>]
let splitProps (o: 'T, arr1: string[], arr2: string[]) : 'T * 'T = jsNative

[<Erase>]
type Select() =
    inherit select()

    [<SolidTypeComponent>]
    member props.__ =
        let rootProps, selectProps =
            splitProps (props, [| "name"; "placeholder"; "required"; "disabled" |], [| "placeholder"; "ref"; "onInput"; "onChange"; "onBlur" |])

        div().spread rootProps { select().spread selectProps }
