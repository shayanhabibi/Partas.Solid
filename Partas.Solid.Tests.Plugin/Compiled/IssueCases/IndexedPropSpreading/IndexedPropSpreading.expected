import { splitProps } from "solid-js";

export function Select(props) {
    const [PARTAS_LOCAL, PARTAS_OTHERS] = splitProps(props, []);
    const patternInput = splitProps(props, ["name", "placeholder", "required", "disabled"], ["placeholder", "ref", "onInput", "onChange", "onBlur"]);
    return <div {...patternInput[0]} bool:n$={false}>
        <select {...patternInput[1]} bool:n$={false} />
    </div>;
}

